SU2_CFD/src/numerics/turbulent/turb_sources.cpp

```diff
@@ -76,6 +76,179 @@ CNumerics::ResidualType<> CSourcePieceWise_TurbSA::ComputeResidual(const CConfig
 //  AD::SetPreaccIn(TurbVar_Grad_i[0], nDim);
 //  AD::SetPreaccIn(Volume); AD::SetPreaccIn(dist_i);
 
+//  BC Transition Model variables
+  su2double vmag, rey, re_theta, re_theta_t, re_v;
+  su2double tu , nu_cr, nu_t, nu_BC, chi_1, chi_2, term1, term2, term_exponential;
+
+  // Set the boolean here depending on whether the point is closest to a rough wall or not.
+  roughwall = (roughness_i > 0.0);
+
```

Comment:
For something called "no ft2" this looks like wayyyyy too much code (correct me if I'm wrong, I did not do a diff).
I know for these classes we operated on a "copy first, ask questions later" basis, but that has stopped. The duplication police has entered the building :)

Reply:
It is a problem with the trace. At some moment I had to start from scratch so I deleted everything and copy pasted all code on turb_sources.cpp, and implemented the modifications.